src/Data/Newtype.purs

```diff
@@ -17,6 +43,17 @@ ala
   -> f a
 ala _ f = map unwrap (f wrap)
 
+-- | Similar to `ala` but useful for cases where you want to use an additional
+-- | projection with the higher order function:
+-- |
+-- | ``` purescript
+-- | ala' Additive foldMap String.length ["hello", "world"] -- 10
```

Comment:
Actually, I'll move all of the examples into a tests file so they're at least typechecked, etc. with stubs for the "external" functions.